components/match2/wikis/heroes/legacy/match_group_legacy_default.lua

```diff
@@ -0,0 +1,192 @@
+---
+-- @Liquipedia
+-- wiki=heroes
+-- page=Module:MatchGroup/Legacy/Default
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchGroupLegacyDefault = {}
+
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+
+local MAX_NUMBER_OF_OPPONENTS = 2
+local MAX_NUM_MAPS = 20
+
+local roundData
+---@param templateid string
+---@param bracketType string
+---@return table
+function MatchGroupLegacyDefault.get(templateid, bracketType)
+	local lowerHeader = {}
+	local matches = mw.ext.Brackets.getCommonsBracketTemplate(templateid)
+
+	assert(type(matches) == 'table')
+	local bracketData = {}
+	roundData = roundData or {}
+	local lastRound = 0
+	for _, match in ipairs(matches) do
```

Comment:
maybe Array.forEach

Reply:
We are always duplicating code in this module, maybe we could move more functions to base module

Reply:
imo okay to move parts of it to commons as long as we can still overwrite stuff that needs overwrites on some wikis

Reply:
I will take a better look at that possibility when time allows :P